client/session/profile_manager.go

```diff
@@ -0,0 +1,242 @@
+package session
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	// SpaceliftConfigDirectory is the name of the Spacelift config directory.
+	SpaceliftConfigDirectory = ".spacelift"
+
+	// CurrentFileName is the name of the symlink that points at the current profile.
+	CurrentFileName = "current"
+)
+
+// invalidProfileAliases contains a list of strings that cannot be used as profile aliases.
+var invalidProfileAliases = []string{"/", "\\", "current", ".", ".."}
+
+// A Profile represents a spacectl profile which is used to store credential information
+// for accessing Spacelift.
+type Profile struct {
+	// The alias (name) of the profile.
+	Alias string `json:"alias,omitempty"`
+
+	// The credentials used to make Spacelift API requests.
+	Credentials *StoredCredentials `json:"credentials,omitempty"`
+}
+
+// A ProfileManager is used to interact with Spacelift profiles.
+type ProfileManager struct {
+	// The directory that profiles are stored in.
+	ProfilesDirectory string
+
+	// The path to the currently selected profile.
+	CurrentPath string
+}
+
+// NewProfileManager creates a new ProfileManager using the specified directory to store the profile data.
+func NewProfileManager(profilesDirectory string) *ProfileManager {
+	return &ProfileManager{
+		ProfilesDirectory: profilesDirectory,
+		CurrentPath:       filepath.Join(profilesDirectory, CurrentFileName),
+	}
+}
+
+// Init initialises the profile manager, making sure it is ready to store and retrieve profiles.
+func (m *ProfileManager) Init() error {
```

Comment:
Would it be more normal to include this logic in `NewProfileManager`?

Reply:
I'm not an ideologue, so no opinion here.
@cube2222 any thoughts?

Reply:
I would move this to the `NewProfileManager`. 
A one line function for setting up the the manager wouldn't need a separate function.

Reply:
No strong opinions. I guess a separate Init is a non-formal contract the user of the ProfileManager must uphold, so one more thing to think about, whereas by moving it into NewProfileManager there's less room for misuse.

Reply:
We could even use `init()` for this 😄

Reply:
**no**

Reply:
It was actually worth doing - I realised when I merged them that I had a place where it wasn't being initialised 🤦